frame/nomination-pools/src/tests.rs

```diff
@@ -5100,3 +5215,1045 @@ mod reward_counter_precision {
 			});
 	}
 }
+
+mod commission {
+	use super::*;
+
+	#[test]
+	fn set_commission_works() {
+		ExtBuilder::default().build_and_execute(|| {
+			// Set a commission for pool 1.
+			assert_ok!(Pools::set_commission(
+				RuntimeOrigin::signed(900),
+				1,
+				Some((Perbill::from_percent(50), 900))
+			));
+
+			let commission_as_percent = BondedPool::<Runtime>::get(1)
+				.unwrap()
+				.commission
+				.current
+				.as_ref()
+				.map(|(x, _)| *x)
+				.unwrap_or(Perbill::zero());
+			assert_eq!(commission_as_percent, Perbill::from_percent(50));
+
+			// update commission only.
```

Comment:
I would expect to see a payout after each variation here in order to fully test this.